File: Leetcode C#/Longest Common Prefix.cs
public class Solution {
    public string LongestCommonPrefix(string[] strs) {

    int minLength = strs.Min(e => e.Length);
    string res = strs.First(e => e.Length == minLength);

    foreach (string str in strs)
    {
        for (int j = 0; j < minLength; j++)
        {
            if (str[j] != res[j])
            {
                minLength = j; 
                break;
               
            }
        }
    }
    if (minLength == 0)
        return "";
    return res.Substring(0, minLength);

        }

    }

File: 14-longest-common-prefix/longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        
        if (strs == null || strs.length == 0)
            return "";

        String prefix = strs[0];
        int prefLen = prefix.length();
        for (int i = 1; i < strs.length; i++) {
            String cur = strs[i];
            while(prefLen>cur.length()||!prefix.equals(cur.substring(0, prefLen))) {
                prefLen--;
                if(prefLen==0)
                    return "";
                prefix=prefix.substring(0, prefLen);
            }
        }
        return prefix;
    }
}